lib/parsers/iso_base_file_format/decoder.rb

```diff
@@ -0,0 +1,1044 @@
+# This class provides generic methods for parsing file formats based on QuickTime-style "atoms", such as those seen in
+# the ISO base media file format (ISO/IEC 14496-12), a.k.a MPEG-4, and those that extend it (MP4, CR3, HEIF, etc.).
+#
+# For more information on atoms, see https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/QTFFChap1/qtff1.html
+
+class FormatParser::Decoder
```

Comment:
As you mentioned in another comment, you implemented a lot of those other atoms because it's a public repo so it would be of interest to others. I think the fact that we added support for those atoms in a public repo warrants having tests for it. 

It's a massive undertaking though, so maybe you could remove what we don't use for now, and only add them when we can actually provide tests for them... I'm just uncomfortable with publishing untested code to a public open-source project...

Reply:
Aight, I'll comment out what's not necessary for CR3 and we can reintroduce them down the line.